node/src/main/scala/coop/rchain/node/diagnostics/effects/package.scala

```diff
@@ -61,26 +76,38 @@ package object effects {
       import A._
       import Trace._
 
+      private val spans: TrieMap[TraceId, SourceTrace] = TrieMap.empty
+
       override def mark(name: String): F[Unit] = {
-        ask.map {
-          case DefaultTrace(source) =>
-            trace(source) {
-              mark(name)
-            }
-          case KamonTrace(ks) => ks.mark(name)
+        ask.flatMap { traceId =>
+          val k: Option[Trace] = spans.get(traceId)
+          k match {
```

Comment:
Using the matches is verbose, less readable and error prone. I think we should always use the appropriate combinators instead:
```scala
spans.get(traceId)
  .map(_.mark(name))
  .getOrElse(Sync[F].unit)
```